prov/tcp/src/tcpx.h

```diff
@@ -149,11 +150,17 @@ struct tcpx_cq_data_hdr {
 			 TCPX_IOV_LIMIT +			\
 			 TCPX_MAX_INJECT_SZ)
 
+typedef void (*tcpx_hdr_ntoh_fn_t)(struct tcpx_base_hdr *hdr);
+typedef void (*tcpx_hdr_hton_fn_t)(struct tcpx_base_hdr *hdr);
+typedef uint64_t (*tcpx_hdr_sz_ntoh_fn_t)(struct tcpx_base_hdr *hdr);
+typedef uint16_t (*tcpx_hdr_flgs_ntoh_fn_t)(struct tcpx_base_hdr *hdr);
```

Comment:
Drop typedefs and just define function pointers directly as part of tcpx_ep structure